lightning/src/routing/network_graph.rs

```diff
@@ -1813,4 +1932,165 @@ mod tests {
+				chain_hash,
+				full_information: true,
+				first_blocknum: 0,
+				number_of_blocks: 2000,
+				short_channel_ids: vec![
+					0x0003e0_000000_0000, // 992x0x0
+					0x0003e8_000000_0000, // 1000x0x0
+					0x0003e9_000000_0000, // 1001x0x0
+					0x0003f0_000000_0000, // 1008x0x0
+					0x00044c_000000_0000, // 1100x0x0
+					0x0006e0_000000_0000, // 1760x0x0
+				],
+			});
+			assert!(result.is_ok());
+
+			// We expect to emit a query_short_channel_ids message with the received scids
+			let events = net_graph_msg_handler.get_and_clear_pending_msg_events();
+			assert_eq!(events.len(), 1);
+			match &events[0] {
+				MessageSendEvent::SendShortIdsQuery { node_id, msg } => {
+					assert_eq!(node_id, &node_id_1);
+					assert_eq!(msg.chain_hash, chain_hash);
+					assert_eq!(msg.short_channel_ids, vec![
+						0x0003e0_000000_0000, // 992x0x0
+						0x0003e8_000000_0000, // 1000x0x0
+						0x0003e9_000000_0000, // 1001x0x0
+						0x0003f0_000000_0000, // 1008x0x0
+						0x00044c_000000_0000, // 1100x0x0
+						0x0006e0_000000_0000, // 1760x0x0
+					]);
+				},
+				_ => panic!("expected MessageSendEvent::SendShortIdsQuery"),
+			}
+		}
+
+		// Test receipt of a reply that indicates the remote node does not maintain up-to-date
+		// information for the chain_hash. Because of discrepancies in implementation we use
+		// full_information=false and short_channel_ids=[] as the signal.
+		{
+			// Handle the reply indicating the peer was unable to fulfill our request.
+			let result = net_graph_msg_handler.handle_reply_channel_range(&node_id_1, ReplyChannelRange {
+				chain_hash,
+				full_information: false,
+				first_blocknum: 1000,
+				number_of_blocks: 100,
+				short_channel_ids: vec![],
+			});
+			assert!(result.is_err());
+			assert_eq!(result.err().unwrap().err, "Received reply_channel_range with no information available");
+		}
+	}
+
+	#[test]
+	fn handling_reply_short_channel_ids() {
+		let (secp_ctx, net_graph_msg_handler) = create_net_graph_msg_handler();
+		let node_privkey = &SecretKey::from_slice(&[41; 32]).unwrap();
+		let node_id = PublicKey::from_secret_key(&secp_ctx, node_privkey);
+
+		let chain_hash = genesis_block(Network::Testnet).header.block_hash();
+
+		// Test receipt of a successful reply
+		{
+			let result = net_graph_msg_handler.handle_reply_short_channel_ids_end(&node_id, ReplyShortChannelIdsEnd {
+				chain_hash,
+				full_information: true,
+			});
+			assert!(result.is_ok());
+		}
+
+		// Test receipt of a reply that indicates the peer does not maintain up-to-date information
+		// for the chain_hash requested in the query.
+		{
+			let result = net_graph_msg_handler.handle_reply_short_channel_ids_end(&node_id, ReplyShortChannelIdsEnd {
+				chain_hash,
+				full_information: false,
+			});
+			assert!(result.is_err());
+			assert_eq!(result.err().unwrap().err, "Received reply_short_channel_ids_end with no information");
+		}
+	}
+
+	#[test]
+	fn handling_query_channel_range() {
```

Comment:
What does this test and the following are checking ? Are they still accurate given last modifications ?

Reply:
These two tests are for the stubbed method for handling inbound queries.  Since they're stubbed they just test that it results in an Err.

Reply:
Ah yes true.